pkg/extensions/search/resolver.go

```diff
@@ -216,39 +216,45 @@ func globalSearch(repoList []string, name, tag string, olu common.OciLayoutUtils
 			log.Error().Err(err).Msgf("can't find latest updated tag for repo: %s", repo)
 		}
 
-		tagsInfo, err := olu.GetImageTagsWithTimestamp(repo)
+		repoInfo, err := olu.GetExpandedRepoInfo(repo)
 		if err != nil {
-			log.Error().Err(err).Msgf("can't get tags info for repo: %s", repo)
+			log.Error().Err(err).Msgf("can't get repo info for repo: %s", repo)
 
 			continue
 		}
 
-		repoInfo, err := olu.GetExpandedRepoInfo(repo)
+		manifests, err := olu.GetImageManifests(repo)
 		if err != nil {
-			log.Error().Err(err).Msgf("can't get repo info for repo: %s", repo)
+			log.Error().Err(err).Msgf("can't get manifests for repo: %s", repo)
+
+			continue
+		}
+
+		if len(manifests) != len(repoInfo.Manifests) {
+			log.Error().Msgf("number of manifests incorrect %d, expected %d", len(manifests), len(repoInfo.Manifests))
 
 			continue
 		}
 
 		var lastUpdatedImageSummary gql_generated.ImageSummary
 
-		repoPlatforms := make([]*gql_generated.OsArch, 0, len(tagsInfo))
+		repoPlatforms := make([]*gql_generated.OsArch, 0, len(manifests))
 		repoVendors := make([]*string, 0, len(repoInfo.Manifests))
 
 		for i, manifest := range repoInfo.Manifests {
 			imageLayersSize := int64(0)
 
-			imageBlobManifest, err := olu.GetImageBlobManifest(repo, godigest.Digest(tagsInfo[i].Digest))
+			imageBlobManifest, err := olu.GetImageBlobManifest(repo, manifests[i].Digest)
```

Comment:
Why use `manifests[i].Digest` instead of using the digest from the `manifest` variable. I don't think there's a guarantee `repoInfo.Manifests` are in the same order as in `manifests`.
This was probably an issue before, but let's fix it now since we found it.

Reply:
Main logic is now changed, we use only a list of manifests.